.github/workflows/install-from-condalock-file.yml

```diff
@@ -0,0 +1,104 @@
+# Install esmvaltool from a conda lock file
+# To build a conda lock file install conda-lock first then
+# run conda-lock lock --platform linux-64 -f environment.yml --mamba
+# (mamba activated for speed). Change platform for osx-64 or win-64.
+# Env creation then happens as per normal use with
+# conda create --name esmvaltool-fromlock --file conda-linux-64.lock
+# note that pip and conda are NOT installed.
+
+name: Conda-lock Install
+on:
+  push:
+    branches:
+    - main
+    - condalock  # just testing
+  # run the test only if the PR is to main
+  # turn it on if required
+  #pull_request:
+  #  branches:
+  #  - main
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  linux:
+    runs-on: "ubuntu-latest"
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+      fail-fast: false
+    name: Linux Python ${{ matrix.python-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: esmvaltool-fromlock
+          # environment-file: conda-linux-64.lock
+          python-version: ${{ matrix.python-version }}
+          miniconda-version: "latest"
+          channels: conda-forge
+      - shell: bash -l {0}
+        run: mkdir -p source_install_linux_artifacts_python_${{ matrix.python-version }}
+      - shell: bash -l {0}
+        run: conda --version 2>&1 | tee source_install_linux_artifacts_python_${{ matrix.python-version }}/conda_version.txt
+      - shell: bash -l {0}
+        run: which python
+      - shell: bash -l {0}
+        run: python -V 2>&1 | tee source_install_linux_artifacts_python_${{ matrix.python-version }}/python_version.txt
+      - shell: bash -l {0}
+        run: conda create --name esmvaltool-fromlock --file conda-linux-64.lock
+      - shell: bash -l {0}
+        run: conda install pip
+      - shell: bash -l {0}
+        run: which python
+      - shell: bash -l {0}
+        run: pip --version
+      - shell: bash -l {0}
+        run: pip install -e .[develop]
+      - shell: bash -l {0}
+        run: esmvaltool --help
+      - shell: bash -l {0}
+        run: esmvaltool version 2>&1 | tee source_install_linux_artifacts_python_${{ matrix.python-version }}/version.txt
+      - shell: bash -l {0}
+        run: pytest -n 2 -m "not installation"
```

Comment:
Similar to above, do we need every single line in its own bash?

Reply:
I forgot to add above - we could use the `run: | ` syntax, and use the bash spec in one place only, but that'll mask the commands under a single name (whatever your `- name:` argument will be)

Reply:
Potaeto Potahto; you say mask, I say group. I think collecting the commands in few named shells would clarify, not obscure. Perhaps something like `prepare`, `install`, `check`, `test`?

Reply:
sure, sounds like a good plan - for the next Github Action that @schlunma as maintainer in chief after v2.5.0 will write, I wrote all the others with bashing around, so that'd mean changing all the other actions too - as Consuela from Family Guy would say: nooooo :laughing:

Reply:
Of course it wouldn't mean that and just because something bad happened in the past doesn't mean that we have to repeat it in the present or future. This is a very easy and simple change. If it has merit, it should be done, if not not.

Reply:
agreed, but I personally really like all those steps nicely laid out and shown when the thing runs, I like more stuff on screens than less stuff on screens :blush: Unlike others like @bouweandela that cut output like weeds :rofl: